fem/nonlininteg.cpp

```diff
@@ -10,10 +10,35 @@
 // Software Foundation) version 2.1 dated February 1999.
 
 #include "fem.hpp"
+#include "../general/forall.hpp"
```

Comment:
Should we split the kernels into separate files by integrator (e.g. `nonlininteg_convection.cpp`) as we do for bilinear form integrators?

Reply:
I think we should do that, but I'd like to hear an opinion from @tzanio and @v-dobrev.

Reply:
I agree, it makes sense to separate the kernels.

Reply:
agreed also